cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java

```diff
@@ -107,16 +112,23 @@ public UpdateResult<ProvenanceWorkflowRun<ExternalKey>> update() throws Exceptio
       EPOCH.labels(baseUrl).set(body.getEpoch());
       TIMESTAMP.labels(baseUrl).set(body.getTimestamp());
       ERROR.labels(baseUrl).set(0);
-      for (final var workflowRun : body.getResults()) {
-        workflowRun.setInstanceName(instanceName);
-      }
+      List<ProvenanceWorkflowRun<ExternalKey>> validBodyResults = new ArrayList<>();
+      body.getResults().stream()
+          .filter(
+              workflowRun ->
+                  workflowRun.getExternalKeys().stream()
+                      .noneMatch(
+                          externalKey -> ignoreProviders.contains(externalKey.getProvider())))
+          .peek(workflowRun -> workflowRun.setInstanceName(instanceName))
```

Comment:
Why did you choose to use `peek` here?

Reply:
Originally I had ` .map(workflowRun -> { workflowRun.setInstanceName(instanceName); return workflowRun; })` and intellij suggested to "replace with peek". 
From a google looks like that's bad practice, so I will change